packages/typir/src/dependency-injection.ts

```diff
@@ -0,0 +1,157 @@
+/******************************************************************************
+ * Copyright 2024 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/**
```

Comment:
Just as an idea for the future: Since we use this DI functionality and some more utility functions like `assertUnreachable` at least in Langium and Typir, maybe we should extract that into a new utility library, something like `dsl-utils` or `typefox-utils`?